modules/javafx.controls/src/main/java/javafx/scene/control/skin/MenuBarSkin.java

```diff
@@ -927,10 +946,11 @@ private void rebuildUI() {
             }
         }
 
-        getSkinnable().focusedProperty().addListener(menuBarFocusedPropertyListener);
+        getSkinnable().focusedProperty().addListener(weakMenuBarFocusedPropertyListener);
+
```

Comment:
- listeners on skinnable's properties should be installed via skin api
- skinnable's properties don't change, so no need to add/remove in rebuild - instead do it once

take both with a grain of salt, though, and test if that's really possible :) I think it does - the value is a boolean so the old is always the !current.